### DISCUSSION TITLE
Howdo disable IPV6

### DISCUSSION THREAD

**Discussion by DittelHome** [arendst/Tasmota#21380] [Q&A]
Hi Experts,
is it possible to disable IPV6 on my ESP32-c3. (Firmware build with gitpod)
I only found:
IPv6 is enabled by default on ESP32 since v12.3.1.1.
regards Klaus

**Comment by s-hadinger** [comment]
Any reason why you want to disable it?

If so you need to undefine `USE_IPV6`

**Comment by DittelHome** [comment]
I added this to my user_config_override.h:

```
ifdef USE_IPV6
#undef USE_IPV6
#endif
```
It works like a charm. THANK YOU !!

No specific reason, IMHO iot devices does not need IPV6.  Maybe a security reason ..
The ESP32 is used for SmartMeter ..

**Comment by DittelHome** [comment]
OK, after disabling the IPV6, stuff under menu  Information is gone.  
But the device still have a ULA adress:
`SmartMeter.fritz.box (192.168.1.18, fd00::4a31:b7ff:feab:ceec)`
Do i have to disable another thing ?

ping fd00::4a31:b7ff:feab:ceec  (ping from my laptop)

```
ping wird ausgeführt für fd00::4a31:b7ff:feab:ceec mit 32 Bytes Daten:
Antwort von fd00::4a31:b7ff:feab:ceec: Zeit=176ms
Antwort von fd00::4a31:b7ff:feab:ceec: Zeit=5ms
Antwort von fd00::4a31:b7ff:feab:ceec: Zeit=6ms
Antwort von fd00::4a31:b7ff:feab:ceec: Zeit=5ms
```

**Reply by s-hadinger** [reply]
I don't know, but keep in mind that even if you disable IPv6 from Tasmota, it is still present in the underlying LwIP stack.

Anyways, this address is [Unique Local Address](https://en.wikipedia.org/wiki/Unique_local_address):
"These addresses are non-globally reachable (routable only within the scope of private networks, but not the global IPv6 Internet)"

So there shouldn't be any security concern

**Reply by DittelHome** [reply]
Yes, i agree 100%.
The global adress is no not used anymore, therefore all is fine.
